utilities/transactions/write_unprepared_txn_db.cc

```diff
@@ -367,29 +365,49 @@ Iterator* WriteUnpreparedTxnDB::NewIterator(const ReadOptions& options,
   // TODO(lth): Refactor so that this logic is shared with WritePrepared.
   constexpr bool ALLOW_BLOB = true;
   constexpr bool ALLOW_REFRESH = true;
-  std::shared_ptr<ManagedSnapshot> own_snapshot = nullptr;
   SequenceNumber snapshot_seq;
   SequenceNumber min_uncommitted = 0;
-  if (options.snapshot != nullptr) {
-    snapshot_seq = options.snapshot->GetSequenceNumber();
-    min_uncommitted =
-        static_cast_with_check<const SnapshotImpl, const Snapshot>(
-            options.snapshot)
-            ->min_uncommitted_;
-  } else {
-    auto* snapshot = GetSnapshot();
-    // We take a snapshot to make sure that the related data in the commit map
-    // are not deleted.
-    snapshot_seq = snapshot->GetSequenceNumber();
-    min_uncommitted =
-        static_cast_with_check<const SnapshotImpl, const Snapshot>(snapshot)
-            ->min_uncommitted_;
-    own_snapshot = std::make_shared<ManagedSnapshot>(db_impl_, snapshot);
+  // TODO(lth): Currently, we can only create iterators with the transaction
+  // snapshot. This is because the iterator logic relies on snapshot validation
+  // to ensure that no committed keys exist between snapshot_seq < seq <
+  // max_visible_seq. This assumption is currently upheld by snapshot
+  // validation, and validation is run against the transaction snapshot sequence
+  // number.
+  //
```

Comment:
I am thinking this comment might be more clear:
// The greedy approach of sequential scan in iterator's ::Prev implementation breaks if there are writes to the key AFTER the and BEFORE our own uncommitted writes. Such cases are already prevented by ValidateSnapshot logic only if the txn has set a snapshot before doing writes.

Reply:
I think I'll call it something other than "greedy approach", but I'll use the rest of your comment.